sample/trace_key_test.go

```diff
@@ -160,3 +162,341 @@ func createTestTrace(t *testing.T, spans []testSpan) *types.Trace {
 
 	return trace
 }
+
+func TestDistinctValue_AddAsString(t *testing.T) {
+	fields := []string{"field1", "field2"}
+	dv := &distinctValue{
+		buf: make([]byte, 0, 1024),
+	}
+	dv.init(fields, 10)
+
+	tests := []struct {
+		name        string
+		value       any
+		fieldIdx    int
+		wantSuccess bool
+		wantCount   int
+	}{
+		{
+			name:        "Add string",
+			value:       "test-value",
+			fieldIdx:    0,
+			wantSuccess: true,
+			wantCount:   1,
+		},
+		{
+			name:        "Add integer",
+			value:       42,
+			fieldIdx:    0,
+			wantSuccess: true,
+			wantCount:   2,
+		},
+		{
+			name:        "Add float",
+			value:       3.14,
+			fieldIdx:    0,
+			wantSuccess: true,
+			wantCount:   3,
+		},
+		{
+			name:        "Add boolean",
+			value:       true,
+			fieldIdx:    0,
+			wantSuccess: true,
+			wantCount:   4,
+		},
+		{
+			name:        "Add to second field",
+			value:       "second-field",
+			fieldIdx:    1,
+			wantSuccess: true,
+			wantCount:   5,
+		},
+		{
+			name:        "Add duplicate string",
+			value:       "test-value",
+			fieldIdx:    0,
+			wantSuccess: false, // Already exists
+			wantCount:   5,     // Unchanged
+		},
+		{
+			name:        "Add nil",
+			value:       nil,
+			fieldIdx:    0,
+			wantSuccess: false,
+			wantCount:   5, // Unchanged
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got := dv.AddAsString(tt.value, tt.fieldIdx)
+			require.Equal(t, tt.wantSuccess, got)
+			require.Equal(t, tt.wantCount, dv.totalUniqueCount)
```

Comment:
This really isn't testing much. Maybe these cases could be merged into the other big test, with more robust checks on the results?